libsolidity/analysis/TypeChecker.cpp

```diff
@@ -299,7 +299,7 @@ void TypeChecker::checkContractIllegalOverrides(ContractDefinition const& _contr
 					overriding->isPayable() != function->isPayable() ||
 					overridingType != functionType
```

Comment:
I wonder if it would make sense to merge this conditional with `overrideTypeError`.

The problem right now: it is in two places and need to be kept in sync.

The problem if merged: the conditional might be too far from this location (for readability purposes).